core/src/main/scala/cats/sequence/sequence.scala

```diff
@@ -17,85 +17,157 @@ trait Sequencer[L <: HList] extends Serializable {
 trait RecordSequencer[L <: HList] extends Serializable {
   type Out
+
   def apply(record: L): Out
 }
 
-object RecordSequencer {
-  type Aux[L <: HList, Out0] = RecordSequencer[L] { type Out = Out0 }
+trait RecordSequencerInstanceLowPrio {
+  type Aux[L <: HList, Out0] = RecordSequencer[L] {type Out = Out0}
 
+  implicit def mkRecordSequencerForInvariant[R <: HList, K <: HList, V <: HList, F[_], VOut <: HList, ZOut <: HList](
+                                                                                                                      implicit unzip: UnzipFields.Aux[R, K, V],
+                                                                                                                      valueSequencer: Sequencer.Aux[V, F, VOut],
+                                                                                                                      F: Invariant[F],
+                                                                                                                      zip: ZipWithKeys.Aux[K, VOut, ZOut],
+                                                                                                                      values: Values.Aux[ZOut, VOut]
+                                                                                                                    ): RecordSequencer.Aux[R, F[ZOut]] = new RecordSequencer[R] {
+    type Out = F[zip.Out]
+
+    def apply(record: R) = F.imap(valueSequencer(unzip.values(record)))(zip(_))(values(_))
+  }
+
+}
+
+object RecordSequencer extends RecordSequencerInstanceLowPrio {
   implicit def mkRecordSequencer[R <: HList, K <: HList, V <: HList, F[_], VOut <: HList](
-    implicit unzip: UnzipFields.Aux[R, K, V],
-    valueSequencer: Sequencer.Aux[V, F, VOut],
-    F: Functor[F],
-    zip: ZipWithKeys[K, VOut]
-  ): RecordSequencer.Aux[R, F[zip.Out]] = new RecordSequencer[R] {
+                                                                                           implicit unzip: UnzipFields.Aux[R, K, V],
+                                                                                           valueSequencer: Sequencer.Aux[V, F, VOut],
+                                                                                           F: Functor[F],
+                                                                                           zip: ZipWithKeys[K, VOut]
+                                                                                         ): RecordSequencer.Aux[R, F[zip.Out]] = new RecordSequencer[R] {
     type Out = F[zip.Out]
+
     def apply(record: R) = F.map(valueSequencer(unzip.values(record)))(zip(_))
   }
 }
 
 trait GenericSequencer[L <: HList, T] extends Serializable {
   type Out
+
   def apply(hl: L): Out
 }
 
-object GenericSequencer {
-  type Aux[L <: HList, T, Out0] = GenericSequencer[L, T] { type Out = Out0 }
+trait GenericSequencerInstanceLowPrio {
```

Comment:
This is to ensure we can evolve it later while preserving binary compatibility:
```suggestion
private[sequence] abstract class GenericSequencerInstanceLowPrio {
```